File: Java/048.java
import java.math.BigInteger;

public class Problem048 {

    public static void main(String[] args) {
        BigInteger sum = BigInteger.ZERO;
        BigInteger modulus = BigInteger.TEN.pow(10);

        for (int i = 1; i <= 1000; i++) {
            BigInteger base = BigInteger.valueOf(i);
            BigInteger term = base.pow(i);
            sum = sum.add(term);
            sum = sum.mod(modulus); // Keep the sum within the last 10 digits
        }
        System.out.println(sum);
    }
}

File: Python/048.py
# The series, 1^1 + 2^2 + 3^3 + ... + 10^{10} = 10405071317.
# Find the last ten digits of the series, 1^1 + 2^2 + 3^3 + ... + 1000^{1000}.


def digits_self_powers(bound):
    res = 0
    for x in range(1, bound + 1):
        res += (x**x) % 10000000000
        res %= 10000000000
    return res


if __name__ == "__main__":
    bound = 1000

    print(digits_self_powers(bound))
